tests/test_dataset.py

```diff
@@ -64,22 +64,53 @@ def test_non_cf_compliant_time_is_decoded(self):
             name="time",
             data=np.array(
                 [
-                    cftime.datetime(2000, 1, 1),
-                    cftime.datetime(2000, 2, 1),
-                    cftime.datetime(2000, 3, 1),
-                    cftime.datetime(2000, 4, 1),
-                    cftime.datetime(2000, 5, 1),
-                    cftime.datetime(2000, 6, 1),
-                    cftime.datetime(2000, 7, 1),
-                    cftime.datetime(2000, 8, 1),
-                    cftime.datetime(2000, 9, 1),
-                    cftime.datetime(2000, 10, 1),
-                    cftime.datetime(2000, 11, 1),
-                    cftime.datetime(2000, 12, 1),
-                    cftime.datetime(2001, 1, 1),
-                    cftime.datetime(2001, 2, 1),
-                    cftime.datetime(2001, 3, 1),
+                    cftime.DatetimeGregorian(
+                        2000, 1, 1, 0, 0, 0, 0, has_year_zero=False
```

Comment:
All of the `cftime.datetime` objects in our tests are now updated to `cftime.<OBJECT_TYPE>`.